src/main/java/org/broadinstitute/hellbender/tools/spark/sv/AlignedAssemblyOrExcuse.java

```diff
@@ -122,21 +117,32 @@ public FermiLiteAssembly getAssembly() {
      */
     static void writeSAMFile( final String samFile,
                                      final SAMFileHeader header,
-                                     final List<AlignedAssemblyOrExcuse> alignedAssemblyOrExcuseList ) {
-        try ( final OutputStream os = BucketUtils.createFile(samFile) ) {
-            final SAMTextWriter writer = new SAMTextWriter(os);
-            writer.setSortOrder(SAMFileHeader.SortOrder.queryname, true);
-            writer.setHeader(header);
-
+                                     final List<AlignedAssemblyOrExcuse> alignedAssemblyOrExcuseList, final boolean preOrdered ) {
+        try ( final SAMFileWriter writer = createSAMFileWriter(samFile, header, preOrdered) ) {
```

Comment:
Not sure if you have run this branch on the cluster, but I am not sure if this could write to HDFS as master currently does. Preferably a test with MiniCluster could test if it works (we have an example in `FindBreakpointEvidenceSparkIntegrationTest`).